roles/jd-server/src/lib/job_declarator/message_handler.rs

```diff
@@ -38,6 +41,31 @@ impl JobDeclaratorDownstream {
     }
 }
 
+pub fn clear_declared_mining_job(
+    mining_job: DeclareMiningJob,
+    mempool: Arc<Mutex<JDsMempool>>,
+) -> Result<(), Error> {
+    // If there is an old declared mining job, remove its transactions from the mempool
+        // Retrieve necessary data from the old job
+        let transactions_to_remove = mining_job.tx_short_hash_list.inner_as_ref();
```

Comment:
Why do you use tx_short_hash_list?
Afaik it doesn't contain TxIds

Reply:
It was the feedback I got from @jbesraa. I used the short hash list before and used the mempool to get the transaction ID from it But the earlier feedback you get suggested I can use it directly